nilearn/tests/test_signal.py

```diff
@@ -178,6 +178,20 @@ def test_butterworth():
     np.testing.assert_almost_equal(out1, out2)
     np.testing.assert_(id(out1) != id(out2))
 
+    # Test check for equal values in critical frequencies
+    sampling = 1
+    low_pass = 2
+    high_pass = 1
```

Comment:
I'm confused: since high_pass < low_pass, I would consider that the critical frequencies are different here ? Why isn't it the case ?

Reply:
> I think [the failing test's parameters](https://github.com/nilearn/nilearn/blob/00084e1aafc49681869de6c83aab3b6450dc7fb0/nilearn/decoding/tests/test_decoder.py#L451-L453) are now being correctly flagged. With the requested band-pass filter of 2-1 Hz and the provided TR of 1s, we're only letting through signal occurring at a 0.5-1s timescale, but because 0.5 overlaps with the Nyquist frequency for this sampling rate, it's getting coerced to ~1.

To expand on what Elizabeth describes above,  the passed values for high_pass and low_pass are manipulated inside `butterworth` by a call to `_check_wn`. This check changes each frequency element to freq / float(nyq), but given certain conditions it results in a value just below 1. If this happens for both elements then they are set to an equal value.